packages/contractkit/src/identity/metadata.ts

```diff
@@ -34,19 +35,41 @@ export class IdentityMetadataWrapper {
     })
   }
 
-  static async fetchFromURL(url: string) {
+  static async fetchFromURL(url: string, kit: ContractKit) {
     const resp = await fetch(url)
     if (!resp.ok) {
       throw new Error(`Request failed with status ${resp.status}`)
     }
-    return this.fromRawString(await resp.text())
+    return this.fromRawString(await resp.text(), kit)
   }
 
-  static fromFile(path: string) {
-    return this.fromRawString(readFileSync(path, 'utf-8'))
+  static fromFile(path: string, kit: ContractKit) {
+    return this.fromRawString(readFileSync(path, 'utf-8'), kit)
   }
 
-  static fromRawString(rawData: string) {
+  static async verifySigner(kit: ContractKit, hash: any, signature: any, metadata: any) {
+    // First try to verify on account's address
+    if (!verifySignature(hash, signature, metadata.address)) {
+      // If this fails, signature may still be one of `address`' signers
+      const accounts = await kit.contracts.getAccounts()
+      if (await accounts.isAccount(metadata.address)) {
+        const signers = [
```

Comment:
It would be faster to use Promise.all here, otherwise you are making these calls serially